### ISSUE TITLE
How to transfer controlnet to your own model: Update readme

### ISSUE THREAD

**Issue by specblades** [Mikubill/sd-webui-controlnet#35]
Please, add more detailed tutorial how to transfer controlnet to your own model

**Comment by 2blackbar** [comment]
I think it has built in SD1.5 + its own weights for features detection, but it ignores SD1.5 and uses your modl from get go when you use this repo code...
What i would like instead is separating feature detection weights from SD1.5 so the file is smaller loads faster and then it would work the same as its working now ( isuppose) It kinda loads SD1.5 twice when you use it on plain 1.5 model from what i read

**Comment by Mikubill** [comment]
> separating feature detection weights from SD1.5

use https://github.com/Mikubill/sd-webui-controlnet/blob/main/extract_controlnet.py

**Comment by nasenbaer43** [comment]
Unless I don' tunderstand it: the script extracts the controlnet from the 5.7 GB files (see https://huggingface.co/lllyasviel/ControlNet/tree/main/models). I guess the resulting files will be quite smaller?

Is there any chance that someone who has already downloaded all these models to please upload the extracted results somewhere?

Seems like a bit of "waste" for everyone to download 7 * 5.7 GB.

**Comment by 2blackbar** [comment]
WORKED !!! Thanks a lot !!! 
For anyone else , make bat file and place in models directory, then run it, bat looks like this inside , just change path to yours and  change names accordingly
 python  extract_controlnet.py  --src F:\sd\extensions\sd-webui-controlnet\models\control_sd15_canny.pth --dst F:\sd\extensions\sd-webui-controlnet\models\control_canny.pth

**Comment by 2blackbar** [comment]
> Unless I don' tunderstand it: the script extracts the controlnet from the 5.7 GB files (see https://huggingface.co/lllyasviel/ControlNet/tree/main/models). I guess the resulting files will be quite smaller?
> 
> Is there any chance that someone who has already downloaded all these models to please upload the extracted results somewhere?
> 
> Seems like a bit of "waste" for everyone to download 7 * 5.7 GB.

i posted how to do it its easy, im going to extract more, you can do it on your own and it gives you 1.4gb model not 5.6gb  modl, so it saves VRAM and loads fast

**Comment by nasenbaer43** [comment]
Sorry you misunderstood me. I was talking about the extracted models, aka your 1.4 GB files.
Of course I can download the 5.7 GB files and extract them myself :)

Since you already did it: any chance you can please upload them anywhere?

**Comment by MadLightTheDoggo** [comment]
> WORKED !!! Thanks a lot !!! For anyone else , make bat file and place in models directory, then run it, bat looks like this inside , just change path to yours and change names accordingly python extract_controlnet.py --src F:\sd\extensions\sd-webui-controlnet\models\control_sd15_canny.pth --dst F:\sd\extensions\sd-webui-controlnet\models\control_canny.pth

You have to place the .bat into the directory with extract_controlnet.py and NOT INTO \models\ folder.

**Comment by Hetaneko** [comment]
> 

here i uploaded the trimmed models with safetensors format
https://huggingface.co/Hetaneko/Controlnet-models/tree/main/controlnet_safetensors

**Comment by Echolink50** [comment]
> here i uploaded the trimmed models with safetensors format https://huggingface.co/Hetaneko/Controlnet-models/tree/main/controlnet_safetensors

Sweet. Thanks a lot

**Comment by ClashSAN** [comment]
here's pruned f16 safetensors set - https://huggingface.co/webui/ControlNet-modules-safetensors/tree/main

**Comment by cerarslan** [comment]
Excuse me guys, what is use case of above the models? Just merge our custom models with those?

**Comment by Echolink50** [comment]
> Excuse me guys, what is use case of above the models? Just merge our custom models with those?

No. Just leave your chosen model selected in it's usual location. Load your chosen preprocessor and controlnet model in the drop-down menu

**Comment by GrennKren** [comment]
> here's pruned f16 safetensors set - https://huggingface.co/webui/ControlNet-modules-safetensors/tree/main

**take off sunglasses**
Oh my..

**Comment by deifos** [comment]
@paulo-coronado I'm looking for the same thing how can I merge a controlnet model to my custom one, where you able to get that going?

**Comment by paulo-coronado** [comment]
@deifos Hey brother! You can merge by following this [guide](https://github.com/lllyasviel/ControlNet/discussions/12). Although, the author is not recommending merging that way anymore, because the results are a little bit different. The recommendation is to use A1111 and its API. If you really need to merge them I using the `tool_transfer` is fine!

**Comment by deifos** [comment]
> @deifos Hey brother! You can merge by following this [guide](https://github.com/lllyasviel/ControlNet/discussions/12). Although, the author is not recommending merging that way anymore, because the results are a little bit different. The recommendation is to use A1111 and its API. If you really need to merge them using the `tool_transfer` will make the job nicely!

Thank man, I was able to merge it but how did you use it, did you load it inside the controlnet/models folder or the stable diffusion models folder because it does not seem to have any effect on my generations.

**Comment by paulo-coronado** [comment]
I am not sure if you can use it in A1111 (maybe replacing `control_canny-fp16.safetensors` with your merged model). The way lllyasviel created was to use it in `gradio_pose2image.py` - just replace the following line as the [guide](https://github.com/lllyasviel/ControlNet/discussions/12) suggests:
```Python
model.load_state_dict(load_state_dict('./models/control_any3_openpose.pth', location='cpu'))
```

**Comment by alelordelo** [comment]
> WORKED !!! Thanks a lot !!!
> For anyone else , make bat file and place in models directory, then run it, bat looks like this inside , just change path to yours and change names accordingly
> python extract_controlnet.py --src F:\sd\extensions\sd-webui-controlnet\models\control_sd15_canny.pth --dst F:\sd\extensions\sd-webui-controlnet\models\control_canny.pth

HI @2blackbar and @paulo-coronado ,

I tried now to extract but the  converted.ckpt model has the same size ocustomtrainedCN.ckpt, which is 8.5GB.

Shouldn't this reduce the file size?

reported issue here:
https://github.com/Mikubill/sd-webui-controlnet/issues/895
